sdk/servicebus/azure-servicebus/azure/servicebus/_common/constants.py

```diff
@@ -167,11 +168,11 @@
 MAX_ABSOLUTE_EXPIRY_TIME = 253402243199000
 MESSAGE_STATE_NAME = b"x-opt-message-state"
 
-class ServiceBusReceiveMode(str, Enum):
+class ServiceBusReceiveMode(str, Enum, metaclass=CaseInsensitiveEnumMeta):
     PEEK_LOCK = "peeklock"
     RECEIVE_AND_DELETE = "receiveanddelete"
 
-class ServiceBusMessageState(int, Enum):
+class ServiceBusMessageState(int, Enum, metaclass=CaseInsensitiveEnumMeta):
```

Comment:
@l0lawrence qq: In the guidelines, it says to use this for Enums that also allow strings. Do we also need to specify the case insensitive metaclass when the Enums are not strings?

Reply:
I don't think we should, I will have to specify that in the checker